tests/jobs/test_scheduler_job.py

```diff
@@ -4120,22 +4120,40 @@ def test_task_with_upstream_skip_process_task_instances():
     dr = dag.create_dagrun(run_type=DagRunType.MANUAL, state=State.RUNNING, execution_date=DEFAULT_DATE)
     assert dr is not None
 
+    # def get_ti_from_db(task):
+    #     return (
+    #         session.query(TaskInstance)
+    #         .filter(
+    #             TaskInstance.dag_id == dag.dag_id,
+    #             TaskInstance.task_id == task.task_id,
+    #             TaskInstance.run_id == dr.run_id,
+    #         )
+    #         .one()
+    #     )
+
+    # with create_session() as session:  
+    #     get_ti_from_db(dummy1).state = State.SKIPPED
+    #     get_ti_from_db(dummy2).state = State.SUCCESS
+    #     session.flush()
+   
     with create_session() as session:
         tis = {ti.task_id: ti for ti in dr.get_task_instances(session=session)}
         # Set dummy1 to skipped and dummy2 to success. dummy3 remains as none.
         tis[dummy1.task_id].state = State.SKIPPED
         tis[dummy2.task_id].state = State.SUCCESS
         assert tis[dummy3.task_id].state == State.NONE
+        session.commit()
+        # dr.refresh_from_db(session=session)
 
     # dag_runs = DagRun.find(dag_id='test_task_with_upstream_skip_dag')
```

Comment:
@potiuk 
I tried to get rid of xfail test in `tests/jobs/test_scheduler_job.py::test_task_with_upstream_skip_process_task_instances`  but in this function the comment says it expects dummy3 to be skipped. But it is returned as None. Can you tell me the expected behaviour? 

 I tried to check similar place in code where they try to skip task and find out the state of downstream task but I couldn't find many.

Reply:
not really sure, I am afraid - those are pretty deep dive tests and they might require qutie a bit deeper knowledge (and time to look it up).

Reply:
yes i took time to look but I am confused on this part. Let me try again

Reply:
@potiuk Could i post this in slack and see if i could get some help in fixing this issue?

Reply:
You can try :) but likely in this case there are very few people who know it deeply enough to be able to help (and likely they will have to simply solve it rather than help :) - @ephraimbuddy , myself, @ashb, @uranusjr  are the likely candidates  :)

Reply:
I see. I think I picked up the wrong issue to fix.